packages/synapse-react-client/src/components/ChallengeRegisterButton/ChallengeRegisterButton.tsx

```diff
@@ -1,102 +1,87 @@
-import React, { useEffect, useState } from 'react'
+import React, { useEffect } from 'react'
 import ExitToAppIcon from '@mui/icons-material/ExitToApp'
-import { Box } from '@mui/system'
+import { Box } from '@mui/material'
 import SpinnerButton from '../SpinnerButton/SpinnerButton'
-import { useSynapseContext } from '../../utils'
 import {
   useGetCurrentUserProfile,
   useGetEntityChallenge,
+  useGetIsUserMemberOfTeam,
   useGetUserSubmissionTeams,
 } from '../../synapse-queries'
-import { Challenge, PaginatedIds } from '@sage-bionetworks/synapse-types'
-import { useGetIsUserMemberOfTeam } from '../../synapse-queries/team/useTeamMembers'
-import { SynapseClientError } from '../../utils/SynapseClientError'
+import { SynapseClientError, useSynapseContext } from '../../utils'
 
 export interface ChallengeRegisterButtonProps {
   projectId: string
-  onChallengeError?: (error: SynapseClientError) => void
+  onError?: (error: SynapseClientError) => void
   onJoinClick?: () => void
   onLeaveClick?: () => void
 }
 
-const EMPTY_ID = ''
-
 const ChallengeRegisterButton = ({
   projectId,
-  onChallengeError,
+  onError,
   onJoinClick,
   onLeaveClick,
 }: ChallengeRegisterButtonProps) => {
   const { accessToken } = useSynapseContext()
-  const [challenge, setChallenge] = useState<Challenge>()
-  const { data: userProfile } = useGetCurrentUserProfile()
-  const [isRegistered, setIsRegistered] = useState<boolean>(false)
-  const [hasSubmissionTeam, setHasSubmissionTeam] = useState<boolean>(false)
-  const [loading, setLoading] = useState<boolean>(true)
-  const [requestError, setRequestError] = useState<SynapseClientError>()
```

Comment:
Much of this information could be computed in the function component, instead of keeping track of state variables